src/compiler/binder.ts

```diff
@@ -2986,7 +2986,9 @@ namespace ts {
                 const excludeFlags = SymbolFlags.ValueModuleExcludes & ~SymbolFlags.Assignment;
                 namespaceSymbol = forEachIdentifierInEntityName(entityName, namespaceSymbol, (id, symbol, parent) => {
                     if (symbol) {
-                        addDeclarationToSymbol(symbol, id, flags);
+                        if (!(symbol.flags & SymbolFlags.Alias)) {

```

Comment:
should it also early-exit here? I can't remember how, or if, you can signal this from forEachIdentifierInEntityName.